bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,319 @@
+            url=kata_url
+        )
+        kata_embed.add_field(name="Difficulty", value=kata_information['rank']['name'], inline=False)
+        return kata_embed
+
+    @staticmethod
+    def language_embed(kata_information: dict) -> Embed:
+        """Creates the 'language embed' which displays all languages the kata supports."""
+        languages = '\n'.join(map(str.title, kata_information['languages']))
+        language_embed = Embed(
+            title=f"Supported Languages for {kata_information['name']}",
+            description=f"```nim\n{languages}\n```",
+            color=Colours.python_blue
+        )
+        return language_embed
+
+    @staticmethod
+    def tags_embed(kata_information: dict) -> Embed:
+        """
+        Creates the 'tags embed' which displays all the tags of the Kata.
+
+        Tags explain what the kata is about, this is what codewars.com calls categories.
+        """
+        tags = '\n'.join(kata_information['tags'])
+        tags_embed = Embed(
+            title=f"Tags for {kata_information['name']}",
+            description=f"```nim\n{tags}\n```",
+            color=Colours.grass_green
+        )
+        return tags_embed
+
+    @staticmethod
+    def miscellaneous_embed(kata_information: dict) -> Embed:
+        """
+        Creates the 'other information embed' which displays miscellaneous information about the kata.
+
+        This embed shows statistics such as the total number of people who completed the kata,
+        the total number of stars of the kata, etc.
+        """
+        embed = Embed(
+            title=f"Other Information for {kata_information['name']}",
+            color=Colours.grass_green
+        )
+        embed.add_field(
+            name="`Total Score`",
+            value=f"```css\n{kata_information['voteScore']}\n```",
+            inline=False
+        )
+        embed.add_field(
+            name="`Total Stars`",
+            value=f"```css\n{kata_information['totalStars']}\n```",
+            inline=False
+        )
+        embed.add_field(
+            name="`Total Completed`",
+            value=f"```css\n{kata_information['totalCompleted']}\n```",
+            inline=False
+        )
+        embed.add_field(
+            name="`Total Attempts`",
+            value=f"```css\n{kata_information['totalAttempts']}\n```",
+            inline=False
+        )
+        return embed
+
+    @staticmethod
+    def create_view(dropdown: InformationDropdown, link: str) -> ui.View:
+        """
+        Creates the discord.py View for the Discord message components (dropdowns and buttons).
+
+        The discord UI is implemented onto the embed, where the user can choose what information about the kata they
+        want, along with a link button to the kata itself.
+        """
+        view = ui.View()
+        view.add_item(dropdown)
+        view.add_item(ui.Button(label="View the Kata", url=link))
+        return view
```

Comment:
Should we perhaps add an [`on_timeout`](https://discordpy.readthedocs.io/en/master/api.html#discord.ui.View.on_timeout) that will edit the message to have disabled components?

Compared to the bot's usual trash-can emoji it would be removed, we should signal to the user somehow that the bot is no longer accepting input.

Reply:
UPDATE: This is more complicated than I anticipated so I think we can skip this.

Reply:
This is actually not that complicated. At the end of the command, we want to the wait for the instance of the view to stop.

This is easily achieved with this method: https://discordpy.readthedocs.io/en/master/api.html?highlight=view#discord.ui.View.wait

After this, we can edit the message to remove the view, by running `await message.edit(view=None)`